### PR TITLE
Save and return continued

### PR DESCRIPTION
**Pull request by samuele-mattiuzzo**
Continues the work done on https://digital.dclg.gov.uk/jira/browse/UKRSS-853 and https://digital.dclg.gov.uk/jira/browse/UKRSS-821

FWIW: https://digital.dclg.gov.uk/jira/browse/UKRSS-854 will be done off the back of this one, but in a separate PR

### CODE REVIEW

File: app/controllers/token_based_resume_controller.rb [hunk 1]
```diff
@@ -127,10 +127,16 @@ def generate_application_token(application, magic_link)
...
+    @application_token = ApplicationToken.find_by(unaccompanied_minor: @application)
 
+    if @application_token.present? && Time.zone.now.utc >= (@application_token.created_at + 1.day)
+      # only resend the same link if the application_token has been requested less than a day ago
+      return_link = @application_token.magic_link
+    else
+      personal_uuid = SecureRandom.uuid
+      return_link = generate_magic_link(personal_uuid)
+      generate_application_token(@application, personal_uuid)
+    end
```
Comment (neilkidd) [inline app/controllers/token_based_resume_controller.rb:139]: Do we need to think about cleaning expired tokens?

**Review by neilkidd — APPROVED** [review]
Woohoo :shipit:

File: app/controllers/token_based_resume_controller.rb (hunk 1 above)
Comment (samuele-mattiuzzo) [reply app/controllers/token_based_resume_controller.rb:139]: Possibly yes, but I'll look at it as part of 821 (resend sms token), although was thinking to keep them and add a new functionality later on that cleans up

Comment (neilkidd) [reply app/controllers/token_based_resume_controller.rb:139]: 👍🏻

**Pull request merged**